tox.ini

```
@@ -1,6 +1,10 @@
 [tox]
 envlist =
-    py{38,39,310,3.11,3.12}-django{32,40,41,42}{-sqlite,-mysql,-postgresql},
+    py{310,311,312}-django50-{sqlite,mysql,postgresql}
+    py{38,39,310,311}-django42-{sqlite,mysql,postgresql}
+    py{38,39,310,311}-django41-{sqlite,mysql,postgresql}
+    py{38,39,310}-django40-{sqlite,mysql,postgresql}
+    py{38,39,310}-django32-{sqlite,mysql,postgresql}
     checkqa,
     pylint,
```

You could drop these unnecessary commas whilst you're at it:
```suggestion
    checkqa
    pylint
```